src/change_stream/event.rs

```diff
@@ -0,0 +1,119 @@
+
+    /// For unsharded collections this contains a single field, id, with the value of the id of the
+    /// document updated.  For sharded collections, this will contain all the components of the
+    /// shard key in order, followed by the id if the id isn’t part of the shard key.
+    pub document_key: Option<Document>,
+
+    /// Contains a description of updated and removed fields in this operation.
+    pub update_description: Option<UpdateDescription>,
+
+    /// For operations of type "insert" and "replace", this key will contain the document being
+    /// inserted, or the new version of the document that is replacing the existing
+    /// document, respectively.
+    ///
+    /// For operations of type "update", when the `ChangeStream's` full document type is
+    /// `UpdateLookup`, this key will contain a copy of the full version of the document from
+    /// some point after the update occurred. If the document was deleted since the updated
+    /// happened, it will be `None`.
+    pub full_document: Option<T>,
+}
+
+/// Describes which fields have been updated or removed from a document.
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct UpdateDescription {
+    /// A `Document` containing key:value pairs of names of the fields that were changed, and the
+    /// new value for those fields.
+    pub updated_fields: Document,
+
+    /// An array of field names that were removed from the `Document`.
+    pub removed_fields: Vec<String>,
+}
+
+/// The operation type represented in a given change notification.
+#[derive(Debug, Deserialize, Clone, PartialEq)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub enum OperationType {
+    /// See [insert-event](https://docs.mongodb.com/manual/reference/change-events/#insert-event)
+    Insert,
+
+    /// See [update-event](https://docs.mongodb.com/manual/reference/change-events/#update-event)
+    Update,
+
+    /// See [replace-event](https://docs.mongodb.com/manual/reference/change-events/#replace-event)
+    Replace,
+
+    /// See [delete-event](https://docs.mongodb.com/manual/reference/change-events/#delete-event)
+    Delete,
+
+    /// See [drop-event](https://docs.mongodb.com/manual/reference/change-events/#drop-event)
+    Drop,
+
+    /// See [rename-event](https://docs.mongodb.com/manual/reference/change-events/#rename-event)
+    Rename,
+
+    /// See [dropdatabase-event](https://docs.mongodb.com/manual/reference/change-events/#dropdatabase-event)
+    DropDatabase,
+
+    /// See [invalidate-event](https://docs.mongodb.com/manual/reference/change-events/#invalidate-event)
+    Invalidate,
+}
+
+/// Identifies which collection or database where an event occurred.
+#[derive(Deserialize, Debug)]
+#[serde(untagged)]
+#[non_exhaustive]
+pub enum ChangeStreamEventSource {
+    /// Contains two fields: "db" and "coll" containing the database and collection name in which
+    /// the change happened.
+    Namespace(Namespace),
```

Comment:
I think the documentation here was written with a `Document` in mind as the type for `Namespace`; instead, I think we can simplify this to say something like "The `Namespace` containing the database and collection in which the change occurred."

Reply:
Updated.